lib/settings/pages/settings_page.dart

```diff
@@ -45,6 +51,101 @@ class SettingsPage extends StatelessWidget {
             toolbarHeight: 70.0,
             pinned: true,
           ),
+          SliverToBoxAdapter(
+            child: Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 18),
+              child: SearchAnchor(
+                viewHintText: l10n.search,
+                builder: (BuildContext context, SearchController controller) {
+                  if (Platform.isIOS) {
+                    return CupertinoSearchTextField(
+                      onTap: () {
+                        controller.openView();
+                      },
+                      onChanged: (_) {
+                        controller.openView();
+                      },
+                    );
+                  } else {
+                    return SearchBar(
+                      hintText: l10n.search,
+                      onTap: () {
+                        controller.openView();
+                      },
+                      onChanged: (_) {
+                        controller.openView();
+                      },
+                      leading: const Icon(Icons.search),
+                    );
+                  }
+                },
+                suggestionsBuilder: (BuildContext context, SearchController controller) {
+                  final List<LocalSettings> localSettings = LocalSettings.values
+                      .where((item) => l10n.getLocalSettingLocalization(item.key).toLowerCase().contains(
+                            controller.text.toLowerCase(),
+                          ))
+                      .toSet()
+                      .toList();
+                  localSettings.removeWhere((setting) => setting.key.isEmpty);
+                  localSettings.sortBy((setting) => setting.key);
+
+                  return List<ListTile>.generate(
+                      localSettings.length,
+                      (index) => ListTile(
+                            subtitle: Text(
+                                "${l10n.getLocalSettingLocalization(localSettings[index].category!.toString())} > ${l10n.getLocalSettingLocalization(localSettings[index].subCategory.toString())}"),
+                            onTap: () {
+                              String pageToNav = SETTINGS_GENERAL_PAGE;
+                              switch (localSettings[index].category) {
```

Comment:
A slightly cleaner approach here would be to use switch expressions (really up to you if you'd like to make this change, since its just styling)

e.g.,

```dart
switch (localSettings[index].category) {
  LocalSettingsCategories.posts => pageToNav = SETTINGS_APPEARANCE_POSTS_PAGE,
  LocalSettingsCategories.comments => pageToNav = SETTINGS_APPEARANCE_COMMENTS_PAGE,
  _ => pageToNav = SETTINGS_GENERAL_PAGE,
}
```